utils/weighted_random.go

```diff
@@ -0,0 +1,155 @@
+package utils
+
+import (
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"reflect"
+	"sort"
+	"sync"
+	"time"
+)
+
+var reflectInt = reflect.TypeOf(int(1))
+var reflectStr = reflect.TypeOf("a")
+var reflectFloat = reflect.TypeOf(float64(1))
+
+type WeightedRandom interface {
+	Get() interface{}
+	GetWithSeed(seed string) (interface{}, error)
+}
+
+type Entry struct {
+	Item   interface{}
```

Comment:
nit: add a `type Item interface{}` above

Reply:
In fact, add the SortKey() as a method on Item..

Reply:
not relevant. Removed